processor/src/test/java/org/mapstruct/ap/test/builder/abstractGenericTarget/AbstractTargetMapper.java

```diff
@@ -0,0 +1,39 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.builder.abstractGenericTarget;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface AbstractTargetMapper {
```

Comment:
I'll look into the missed import. As for using the fluent builder pattern, I agree that it would be cool to use it. However, let's take it step by step 😄. First support builders and then we add enhancement for that

Reply:
I can live with that 👍

Reply:
The FQN there is fine. As one `Builder` already exists, so we can't import 2. Maybe we need to do FQN only on top classes see #1386